apps/service_providers/llm_service/runnables.py

```diff
@@ -202,11 +203,14 @@ def _populate_memory(self):
         self.memory.chat_memory.messages = messages
 
     def _save_message_to_history(self, message: str, type_: ChatMessageType):
-        ChatMessage.objects.create(
+        chat_message = ChatMessage.objects.create(
             chat=self.session.chat,
             message_type=type_.value,
             content=message,
         )
+        if not Tag.objects.filter(name=self.experiment.name, team=self.session.team).exists():
```

Comment:
In a multi-bot setup it would make sense to tag these messages with the experiment, but I don't think it makes that much sense in a single bot setup, since that bot is the only one that could have responded.

Reply:
You can pass through a flag from here to indicate if the message should be tagged: https://github.com/dimagi/open-chat-studio/blob/2a4ac2153d48b8ef404580c0c5397ef71d7d5225/apps/chat/bots.py#L84

Maybe something like `tag_message` or `add_experiment_tag`?

Reply:
went with `add_experiment_tag` [e4df433](https://github.com/dimagi/open-chat-studio/pull/440/commits/e4df433afac92faa3fbcf0ba9162c239626e7a42)